### ISSUE TITLE
docs: add examples and a release-aligned rule reference

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#182]
## Parent

- Cross-repository outcome: z-shell/.github#554
- Configured-profile prerequisite: #181
- Wiki implementation: z-shell/wiki#888
- Later discovery architecture: #183

## Objective

Provide the source-owned examples and generated-reference contract needed for
beginner documentation, while keeping stable public docs aligned with the
latest tagged Zsh Lint release.

The wiki currently tells users to install:

```sh
go install github.com/z-shell/zsh-lint/cmd/zsh-lint@latest
```

but its generated reference follows `main`. At the time this issue was opened,
`v1.1.0` does not contain the configured-analysis interface already present in
the generated reference. Stable docs must not silently describe unreleased CLI
behavior.

## Scope

- Correct #181 before presenting configured analysis as stable.
- Add copyable example configurations for:
  - a standalone tool or application; and
  - a plugin with sourced entrypoint, autoload functions, completions, and
    tests.
- Validate examples with repository tests or a dedicated example check.
- Generate a user-facing rule-reference surface without irrelevant Go package,
  import, method, and empty index sections.
- Move the synchronization target to the wiki's dedicated rule-reference page.
- Make stable synchronization release-aware, or visibly version unreleased
  content so `@latest` and public reference behavior cannot drift silently.
- Keep parser-survey and contributor operations outside the beginner path.

## Acceptance criteria

- Copyable examples are valid under configuration schema version 1.
- Generator tests pin readable rule headings, descriptions, examples,
  severities, suppressions, and source revision metadata.
- Workflow-contract tests pin the exact wiki target and generated marker.
- Stable generated content corresponds to a tag, or the page visibly identifies
  its unreleased source revision.
- `go test ./...`, `go vet ./...`, docs-generation, workflow-contract, and
  configured corpus checks pass.
- Wiki issue z-shell/wiki#888 can consume the output without hand-editing the
  generated region.

## Boundary

This issue owns Zsh Lint examples and reference generation. It does not add
source discovery, publish a release, or perform any GitHub publication action.

**Comment by ss-o** [comment]
Tracking is now split by ownership and delivery boundary:

- #181 owns the configured-profile defect and blocks documenting configured
  analysis as stable.
- z-shell/wiki#888 owns the beginner-facing wiki information architecture and
  page implementation.
- #183 owns later source discovery and classification design. It is not part of
  the current documentation implementation.

Active next action: implement #181 locally, then prepare the coordinated
Zsh Lint examples/generator changes and wiki guides. No push, pull request,
release, deployment, or settings mutation is authorized by this progress
record.

**Comment by ss-o** [comment]
Local implementation is prepared and verified.

Changes:

- add validated standalone and plugin example projects;
- test each example's configuration, source-profile resolution, parsing, and finding-free analysis;
- replace the Go API dump with a user-facing rule-reference transformer;
- stamp the reference with the exact published tag and commit;
- generate from the latest published release while using current `main` only for presentation;
- trigger the sync after a successful Release workflow;
- target the wiki's dedicated `08_rule_reference.mdx` page;
- document the stable v1.1.0 versus unreleased `--config` boundary.

Verification:

- `go test ./...`
- `go vet ./...`
- `actionlint .github/workflows/wiki-docs-sync.yml`
- native `zsh -f -n` across every example source
- `git diff --check`

Delivery dependency: merge z-shell/wiki#888 before this workflow change, so the new generated-region target exists when the sync first runs.

No commit, push, or pull request has been created.
